### ISSUE TITLE
Radicale 3.0 breaking changes [storage]

### ISSUE THREAD

**Issue by cinghiopinghio** [39aldo39/Radicale-DecSync#3]
Hi,
Sorry to bring your attention to the fact that with the last version 3.0 of radicale, this plugin is not working anymore.

As per the release notes in radicale: 

> storage plugins require major changes

With the new version `radicale --verify-storage` fails with the following:

```
[2020-05-19 23:13:54 +0200] [8670] [CRITICAL] An exception occurred during storage verification: Failed to load storage module 'radicale_storage_decsync': module 'radicale.storage' has no attribute 'Collection'
```

**Comment by 39aldo39** [comment]
Thanks for your notice!

For now I just added the requirement that Radicale has to be a version older than 3.0, but I will keep this issue open until I upgrade the plugin to support 3.0.

**Comment by cinghiopinghio** [comment]
That's great, Thanks

Excerpts from Aldo Gunsing's message of June 25, 2020 8:23 pm:

**Comment by MaMiMa81** [comment]
Firstly thank you for this project.

My use case is Etar & DecsyncCc on android synced via Syncthing to linux OS with Radicale DecSync & Radicale locally synced via Vdirsyncer to Calcurse.
Its not perfect (moving appointments on both calendars is problematic - have to copy, delete then reinstate at new date/time plus event colours in Etar get re-set to default when syncing from calcurse initially) but trivial matters as far as I'm concerned.

Just upgraded Radicale to 3.0.3 & Radicale Decsync to 1.1.0.

On performing the prescribed:

`radicale --verify-storage`

I get the output:

`[2020-06-27 15:19:46 +0100] [7006] [ERROR] Invalid collection '': discover() takes from 2 to 3 positional arguments but 4 were given
[2020-06-27 15:19:46 +0100] [7006] [CRITICAL] Storage verifcation failed`

However upon testing syncing appears to be performing just as satisfactory as with  Radicale to 2.1.1 & Radicale DecSync  1.0.3.

I haven't adjusted any config files during the upgrade & in the Radicale/NEWS.md (changelog) it states:

> File system storage

    The storage format is compatible with Radicale 2.x.x
    Run radicale --verify-storage to check for errors

So I'm unclear what this means - perhaps I should post over at [https://github.com/Kozea/Radicale/issues](url)?

**Comment by 39aldo39** [comment]
Thanks for the notice! It is fixed now. Note that it only affected `radicale --verify-storage` but not the internal workings.

**Comment by MaMiMa81** [comment]
Yes I did know that you had:

`Fix radicale --verify-storage (#3)`

Thank you 
Was just wondering if you could shed any light on the:

`.....Invalid collection '': discover() takes from 2 to 3 positional arguments but 4 were given......`

Cheers

**Comment by 39aldo39** [comment]
Sure. The `discover()` method only takes 3 arguments for normal use. The file system storage has an optional 4rd argument which is only used in `--verify-storage`. Other storage backends don't have to support it. But as the DecSync backend is based on the file system one, it makes sense to also have support for the `--verify-storage` call.

**Comment by MaMiMa81** [comment]
Thanks for the response  **39aldo39**
 
I posted over Radicale issues & got this reply:

 #https://github.com/Kozea/Radicale/issues/1071#issuecomment-650582646

Although as I have reported back from my testing syncing appears to be performing just as satisfactory as with Radicale to 2.1.1 & Radicale DecSync 1.0.3 I find it a little perplexing your statement:

> But as the DecSync backend is based on the file system one, it makes sense to also have support for the --verify-storage call.

Because as I stated in my original post it fails with a critical error which doesn't look good

Having said that as far as I'm aware it works fine & its really not a problem for me - I am grateful for this project **39aldo39** & once again I would like to thank you.
Cheers

**Comment by 39aldo39** [comment]
But the commit 11ba91e and corresponding release v1.1.1 does fix this issue! If you upgrade to Radicale 3.0.3 & Radicale DecSync 1.1.1, executing `radicale --verify-storage` works without errors.

**Comment by MaMiMa81** [comment]
My apologies - as you see in my original post I updated to release v1.1.0 after clicking the link:

>  39aldo39 added a commit that referenced this issue 2 days ago
@39aldo39
`Update for Radicale version 3 (#3)`
34bcfb4
@39aldo39 39aldo39 closed this 2 days ago

I misread your post:

> 39aldo39 added a commit that referenced this issue 6 hours ago
@39aldo39
`Fix radicale --verify-storage (#3)` 

as your link in your quote just above.

Just upgraded to release v1.1.1 & indeed all's well - thank you.

I'll post over at [https://github.com/Kozea/Radicale/issues/1071](url) to update & correct things.
Cheers
